hamocc/mo_cyano.F90

```diff
@@ -62,46 +62,64 @@ subroutine cyano(kpie,kpje,kpke,kbnd,pddpo,omask,ptho)
 
     ! Local variables
     integer :: i,j,k
-    real    :: oldocetra,dano3
+    real    :: oldocetra,anavail,dansp,dox,dalk
     real    :: ttemp,nfixtfac
 
     intnfix(:,:)=0.0
+
     !
     ! N-fixation by cyano bacteria (followed by remineralisation and nitrification),
+    ! or, for the extended nitrogen cycle only by remin to NH4),
     ! it is assumed here that this process is limited to the mixed layer
     !
     do j=1,kpje
       do i=1,kpie
-        if (omask(i,j).gt.0.5) then
-          do k=1,kmle(i,j)
+        if (omask(i,j) > 0.5) then
+          do k=1,merge(kwrbioz(i,j),kmle(i,j),leuphotic_cya) ! if leuphotic_cya=.true., do bluefix only in euphotic zone
```

Comment:
It was thus far a pragmatic choice to enable investigating its effect when switching between isopycnal and hybrid coordinates. Maybe we can/should keep it until the hybrid coordinates become default. We can make the euphotic zone default, though. Any thoughts @TomasTorsvik ?

Reply:
@JorgSchwinger , for now, I would keep the option and make euphotic zone only for the cyanobacteria (bluefix) the default (which makes scientifically more sense - probably more in line with the aim of NorESM2.3 making it a scientifically used release). I'll do this as last step after the regression testing since it is answer changing.

Reply:
I don't think the vertical grid is a strong argument. I wouldn't expect a big difference anyway: Mixed layers deeper than 100m are only found when it is cold enough and then there is anyway not much N-fixation going on. In very warm waters, the use of euphotic zone might increase N-fixation, but given the low PP we have in these waters that would actually not be so bad.

In summary, I still don't see a good reason to keep the old "mixed layer based N-fixation", but up to you of course.

Reply:
Ok, good plan.